server/src/main/java/org/apache/druid/server/initialization/jetty/ResponseStatusExceptionMapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server.initialization.jetty;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.java.util.common.StringUtils;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+public class ResponseStatusExceptionMapper implements ExceptionMapper<ResponseStatusException>
+{
+  /**
+   * For Response.Status.BAD_REQUEST or Response.Status.NOT_FOUND,
+   * it's suggested to throw exception {@link BadRequestException} or {@link NotFoundException} respectively
```

Comment:
My first thought is that we should apply the same error-handling style for all code, and the `ResponseStatusExceptionManager` maps the exception to a response so that user don't need to care about the `Response` object.

Considering lots of code returns the `Response` object directly, and it seems no gain to change all these code to exception-throwing style, and also seems no one will do this refactoring at any time.

So, I removed this in the newer commit.